crates/viewer/re_space_view_time_series/src/line_visualizer_system.rs

```diff
@@ -205,144 +196,193 @@ impl SeriesLineSystem {
             );
 
             // If we have no scalars, we can't do anything.
-            let Some(all_scalars) = results.get_required_component_dense::<Scalar>(resolver) else {
-                return Ok(());
+            let Some(all_scalar_chunks) = results.get_required_chunks(&Scalar::name()) else {
+                return;
             };
 
-            let all_scalars = all_scalars?;
-
-            let all_scalars_entry_range = all_scalars.entry_range();
-
-            if !matches!(
-                all_scalars.status(),
-                (PromiseResult::Ready(()), PromiseResult::Ready(()))
-            ) {
-                // TODO(#5607): what should happen if the promise is still pending?
-            }
+            let all_scalars_indices = || {
+                all_scalar_chunks
+                    .iter()
+                    .flat_map(|chunk| {
+                        chunk.iter_component_indices(&query.timeline(), &Scalar::name())
+                    })
+                    .map(|index| (index, ()))
```

Comment:
This looks odd. I assume this is related to the requirements for using `range_zip_1x1` below, but maybe hints at the need for some helpers with slightly different shape if it shows up in more places.

Reply:
This only shows up in the time series views which are written in an awkward, `rustc`-pleasing kind of way.

It's still worthy of a comment in any case, you're right.